Source/WebGPU/WebGPU/BindGroup.mm

```diff
@@ -55,6 +55,32 @@ static bool textureViewIsPresent(const WGPUBindGroupEntry& entry)
     if (descriptor.nextInChain)
         return BindGroup::createInvalid(*this);
 
+    if ([m_device argumentBuffersSupport] != MTLArgumentBuffersTier1) {
+        // FIXME: support more than 1 buffer in the bind group
+        auto vertexArgumentBuffer = safeCreateBuffer(sizeof(float*), MTLStorageModeShared);
+        for (uint32_t i = 0; i < descriptor.entryCount; ++i) {
+            const WGPUBindGroupEntry& entry = descriptor.entries[i];
+
+            if (entry.nextInChain)
+                return BindGroup::createInvalid(*this);
+
+            bool bufferIsPresent = WebGPU::bufferIsPresent(entry);
+            bool samplerIsPresent = WebGPU::samplerIsPresent(entry);
+            bool textureViewIsPresent = WebGPU::textureViewIsPresent(entry);
+            if (bufferIsPresent) {
+                id<MTLBuffer> buffer = WebGPU::fromAPI(entry.buffer).buffer();
+                *(float**)vertexArgumentBuffer.contents = (float*)buffer.gpuAddress;
```

Comment:
`gpuAddress` is available from macOS 13 and iOS 16. Do we need to add version guards around it's use or should WebGPU not be compiled for platforms before these?

Reply:
I will add some form of version guard for now and eventually we may likely want to support tier1 argument buffer devices where `gpuAddress` is not needed

Reply:
I guess this is why the build failed?

Reply:
Yeah.